src/Splitter.sol

```diff
@@ -0,0 +1,113 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface VatLike {
+    function move(address, address, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface DaiJoinLike {
+    function vat() external view returns (address);
+    function exit(address, uint256) external;
+}
+
+interface FlapLike {
+    function kick(uint256, uint256) external returns (uint256);
+    function cage(uint256) external;
+}
+
+interface FarmLike {
+    function notifyRewardAmount(uint256 reward) external;
+}
+
+contract Splitter {
+    mapping (address => uint256) public wards;
+
+    VatLike     public immutable vat;
+    DaiJoinLike public immutable daiJoin;
+    FarmLike    public immutable farm;
+
+    FlapLike public flapper;
+    uint256  public burn; // in WAD; 1 WAD = 100% burn
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(
+        address _daiJoin,
+        address _farm
+    ) {
+        daiJoin = DaiJoinLike(_daiJoin);
+        vat     = VatLike(daiJoin.vat());
+        farm    = FarmLike(_farm);
+        
+        vat.hope(_daiJoin);
+        
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "Splitter/not-authorized");
+        _;
+    }
+
+    uint256 internal constant WAD = 10 ** 18;
+    uint256 internal constant RAY = 10 ** 27;
+
+    function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if   (what == "burn") burn = data;
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "flapper") {
+            vat.nope(address(flapper));
+            flapper = FlapLike(data);
+            vat.hope(data);
+        }
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Note: the kick() cadence is determined by flapper.hop()
+    function kick(uint256 tot, uint256) external auth returns (uint256) {
+        vat.move(msg.sender, address(this), tot);
+
+        uint256 lot = tot * burn / WAD;
+        flapper.kick(lot, 0);
+
+        FarmLike farm_ = farm;
+        uint256 pay = (tot - lot) / RAY;
+        DaiJoinLike(daiJoin).exit(address(farm_), pay);
+        farm_.notifyRewardAmount(pay);
```

Comment:
Need to think how the farm's rewards duration can be changed (for example to align to hop), as you can't change it while a program is running. In our case it could be that the spell wants to call `setDuration` but can not execute because `notifyRewardAmount` (called from the Splitter) starves it.

There is a discussion here in the context of the other farms - https://github.com/cantinasec/review-makerdao-endgametoolkit/issues/13

I think a solution that does not require changing the farm code (but would cost a storage read) is to have a `pendingDuration` variable in the Splitter that blocks calling `notifyRewardAmount` if it's not 0.
So we check this variable, if it's 0 we call `notifyRewardAmount` regularly, and if it's not 0 we check if we can call `setDuration` (and reset it to 0), but if we can't do that we don't proceed to `notifyRewardAmount`.

We can think of other solutions as well.

Reply:
Good point. The issue with the `pendingDuration` solution is that the farm has only a single owner (which would be the PauseProxy) so the Splitter would not be able to directly call `setRewardsDuration` (without adding an intermediary owner contract). 

We could also just set `burn = WAD` (and add the condition you mentioned above to skip calling the farm when we burn 100%), but that means the change to the duration would require two spells. Maybe that's still better than implementing a dedicated farm owner contract?

Reply:
Ah interesting about the single owner.
I think two spells should really be a last resort (as it involves a lot of coordination and explaining) and a dedicated owner contract seems a bit ugly indeed.

So think we can change the farm contract for something like this, but better have a single version with endgame-toolkit, so ideally it's a change we can push there. Note that Dewiz already added a reference param there to one of the functions, so this version was already changed by Maker.

One option is to just support multiple owners (though it's a big ugly with the inheritance).

Another option is to change `setRewardsDuration` to be called either by `owner` or by `rewardsDistribution`. For that we need to remove the `onlyOwner` modifier from it and add something like:
`require(msg.sender == owner || msg.sender == rewardsDistribution);`
This imo is a bit nicer sine we extend the special role that already exists, and don't give the splitter a full `owner` status (though some would say that would be more Maker-like).

Finally, we can consider doing the change suggested by Cantina, as it seems that also the regular farms might benefit from it. But that needs to be done carefully, with a thorough review and audit ofc.

Reply:
Created this forked branch with the farm change to `setRewardDuration()` here https://github.com/makerdao/dss-flappers/pull/13. If we agree that this is the way to go, we can port those changes to [StakingRewards.sol](https://github.com/dewiz-xyz/endgame-toolkit/blob/master/src/synthetix/StakingRewards.sol)

Reply:
This is being worked on [here](https://github.com/makerdao/endgame-toolkit/pull/14) so I'm going to resolve.